### ISSUE TITLE
Blockfrost is not reliable

### ISSUE THREAD

**Issue by 34r7h** [cffls/pycardano#61]
A lot of development time has been lost or misappropriated due to inaccuracies from blockfrost. Can we connect with another service?

**Comment by 34r7h** [comment]
Notice the first image has a native asset with quantity of only 1 , whereas the second image is a ss from cardano showing 2.

<img width="414" alt="Screen Shot 2022-07-06 at 23 11 32" src="https://user-images.githubusercontent.com/2566560/177612074-f15efee5-2407-47c9-ba36-a53f8ac63289.png">
<img width="988" alt="Screen Shot 2022-07-06 at 23 10 56" src="https://user-images.githubusercontent.com/2566560/177612107-7b0d16b3-e904-4387-a144-faf82432898a.png">

It's worth noting that my front-end is also pulling from blockfrost (via nodejs) and showing the correct amount..
<img width="408" alt="Screen Shot 2022-07-06 at 23 20 39" src="https://user-images.githubusercontent.com/2566560/177612532-861675d2-bb51-4eb9-8662-eddf3ba4e38f.png">

**Comment by Et9797** [comment]
What seems to be the issue here? As far as I can tell there's two UTxOs, each have the Xymbol token with a quantity of 1. So a total of 2 for the address, as shown on Cardanoscan. What I think you're doing on the front-end is querying Blockfrost's asset addresses endpoint `assets/{asset}/addresses`, however in Pycardano you're querying utxos, so you'll need to sum them up

**Comment by mmahut** [comment]
@34r7h please read a little bit about [how the UTXO model works](https://docs.cardano.org/plutus/eutxo-explainer). The data reported are correct.

**Comment by 34r7h** [comment]
@Et9797 you're right, my mistake. Been a long struggle to get systems working and twice before BF data wasn't updating so I jumped to an incorrect conclusion. Thanks for helping me see.

**Comment by 34r7h** [comment]
@Et9797 i get two utxos from the same tx id when the utxo is a minting tx. difficult to build a change function. can you recommend a good way to handle this scenario:
```
─────────────────────────
utxos_from_bf: <class 'list'>
───────────────

0: {'input': {'index': 0,
 'transaction_id': TransactionId(hex='12a3c696904332e678dc4897b343e111dd688caa9e567e24255aa3a8bd84c754')},
 'output': {'address': addr1q9nzv2662ey4kkx96makz82q99wwzehs2uzrt9wwyttq723xkkkcpe0xgfzya3g0jzz825fyfzwm7melppsjr3uw72qs7am8ys,
 'amount': {'coin': 1379280,
 'multi_asset': {ScriptHash(hex='ef86d15fdc26f796f22582bdafa4369d13e8cd47ef0480b6f57dc899'): {AssetName(b'Panjtara Token'): 1}}},
 'datum_hash': None}} 

1: {'input': {'index': 1,
 'transaction_id': TransactionId(hex='12a3c696904332e678dc4897b343e111dd688caa9e567e24255aa3a8bd84c754')},
 'output': {'address': addr1q9nzv2662ey4kkx96makz82q99wwzehs2uzrt9wwyttq723xkkkcpe0xgfzya3g0jzz825fyfzwm7melppsjr3uw72qs7am8ys,
 'amount': {'coin': 2428467, 'multi_asset': {}},
 'datum_hash': None}} 
```

**Comment by Et9797** [comment]
> @Et9797 i get two utxos from the same tx id when the utxo is a minting tx. difficult to build a change function. can you recommend a good way to handle this scenario:
> 
> ```
> ─────────────────────────
> utxos_from_bf: <class 'list'>
> ───────────────
> 
> 0: {'input': {'index': 0,
>  'transaction_id': TransactionId(hex='12a3c696904332e678dc4897b343e111dd688caa9e567e24255aa3a8bd84c754')},
>  'output': {'address': addr1q9nzv2662ey4kkx96makz82q99wwzehs2uzrt9wwyttq723xkkkcpe0xgfzya3g0jzz825fyfzwm7melppsjr3uw72qs7am8ys,
>  'amount': {'coin': 1379280,
>  'multi_asset': {ScriptHash(hex='ef86d15fdc26f796f22582bdafa4369d13e8cd47ef0480b6f57dc899'): {AssetName(b'Panjtara Token'): 1}}},
>  'datum_hash': None}} 
> 
> 1: {'input': {'index': 1,
>  'transaction_id': TransactionId(hex='12a3c696904332e678dc4897b343e111dd688caa9e567e24255aa3a8bd84c754')},
>  'output': {'address': addr1q9nzv2662ey4kkx96makz82q99wwzehs2uzrt9wwyttq723xkkkcpe0xgfzya3g0jzz825fyfzwm7melppsjr3uw72qs7am8ys,
>  'amount': {'coin': 2428467, 'multi_asset': {}},
>  'datum_hash': None}} 
> ```

Not exactly sure what you're after. If you're just trying to get the balance of an address you can simply use the api function from Pycardano's Blockfrost chain context like so:
```
from pycardano import *

NETWORK = Network.TESTNET
chain_context = BlockFrostChainContext(
    project_id="API_KEY_HERE",
    network=NETWORK
)

nami_address = Address.from_primitive("YOUR_ADDR")

balance = chain_context.api.address(str(nami_address))
print(balance)
```

Now parse the tokens/ADA from the returned Namespace object. Note: the ADA amount is including collateral + Ada locked with assets. So don't be confused if Nami shows a slightly lower amount than that (provided you're using Nami).

**Comment by 34r7h** [comment]
Cheers, @Et9797  

I'm not using Nami, I'm building a wallet interface in my project. Lot's of things work until edge cases come up, such as the token retainers and returns not meeting minimum requirements. It's difficult to reason about this considering there's not much of a difference between a minting transaction and another utxo when balancing inputs and outputs.

Is there a way besides checking via ```blockfrost.BlockFrostApi.asset(asset_policy) ``` if a utxo is a minting utxo containing unspendable ada? I think really this is a problem cardano should address at it's core else I would think many developers will be addressing the same issue continuously.
